examples/model_selection/plot_multiple_metric_evaluation.py

```diff
@@ -0,0 +1,84 @@
+"""Demonstration of multimetric evaluation on cross_val_score and GridSearchCV
+
+Multiple metric grid search (or random search) can be done by setting the
+``scoring`` parameter to a list of metric scorer names or a dict mapping the
+scorer names to the scorer callables.
+
+The scores of all the scorers are available in the ``cv_results_`` dict at keys
+ending in ``'_<scorer_name/metric_name>'`` (``'mean_test_precision'``,
+``'rank_test_precision'``, etc...)
+
+The ``best_estimator_`` and ``best_index_`` attributes are now
+a dict mapping the scorer name to the best estimator and the best candidate
+index corresponding to the best score for that metric which is stored in
+``best_score_[<scorer_name/metric_name>]``.
+
+Similarly the :func:`sklearn.model_selection.cross_val_score`,
+:func:`sklearn.model_selection.learning_curve` and
+:func:`sklearn.model_selection.validation_curve` all support multiple metric
+evaluation when the scoring parameter is a list/tuple of strings denoting
+predefined scorers or a dict mapping the scorer names to the scorer callables.
+"""
+
+# Author: Raghav RV <rvraghav93@gmail.com>
+# License: BSD
+
+import numpy as np
+
+from sklearn.datasets import make_classification
+from sklearn.model_selection import GridSearchCV
+
+from sklearn.metrics import precision_score
+from sklearn.metrics.scorer import make_scorer
+
+from sklearn.svm import SVC
+
+from matplotlib import pyplot as plt
+
+print(__doc__)
+
+
+X, y = make_classification(n_samples=1000, n_features=4, random_state=42)
```

Comment:
In the current rendering, training and test performance is very close for all `C`. We can possibly make them more distinct by increasing the class confusion (increasing `flip_y` or reducing `class_sep`).